var/spack/repos/builtin/packages/python/package.py

```diff
@@ -1154,3 +1155,12 @@ def test(self):
         options = ['-c', 'import sys; {0}'.format(print_str)]
         self.run_test(exe, options=options, expected=[self.spec.prefix],
                       installed=True, purpose=reason)
+
+    @property
+    def cmake_hints(self):
+        """Return standard CMake defines to ensure that the
+        current spec is the one found by CMake find_package(Python, ...)
+        """
+        return [
+            '-DPython_EXECUTABLE={0}'.format(str(self.command))
+        ]
```

Comment:
@adamjstewart Let me know if you're fine with this method. I had to force CMake to find a specific Python external among many, and this is a documented way to do it see: 

https://cmake.org/cmake/help/latest/module/FindPython.html#artifacts-specification

Since the issue seemed potentially relevant for every package depending on Python, I moved the computation of the corresponding defines here.

Reply:
I'm not particularly keen on this. If `cmake_hints` was automatically used by the `CMakePackage` base class that would be one thing, but if you have to manually add it anyway I don't see a huge savings in code duplication here. Also, many packages have multiple different flags for whether to use Python 2 or 3 and which executable to use for both.

Reply:
So would you prefer to move this back to `clingo` or make it automatically used by `CMakePackages` if they depend on Python and Python is external? My understanding is that without something like this using an external Python on a system where multiple versions are installed in the same prefix is very likely prone to error.

Reply:
I'm not familiar enough with CMake. I know they have some kind of standard `FindPython.config` thing or something like that, right? Do all versions of that accept the same CMake flags? In that case, I think it would be fine to add this to the base class, or define hints at the package level and apply them via the base class. Otherwise, I would be fine with moving it back to the `clingo` package.